graph/push_v2.go

```diff
@@ -141,6 +146,13 @@ func (p *v2Pusher) pushV2Tag(tag string) error {
 		dgst, err := p.graph.GetDigest(layer.ID)
 		switch err {
 		case nil:
+			if p.layersPushed[layer.ID] {
```

Comment:
@tiborvass That is probably a reasonable assumption.

It would make more sense to have a download queue that would just fetches of common layers. That is for another day.